client/my-sites/upgrades/domain-management/dns/index.jsx

```diff
@@ -29,6 +30,21 @@ const Dns = React.createClass( {
 		] ).isRequired
 	},
 
+	getInitialState() {
+		return { addNew: true };
+	},
+
+	hasOffice() {
+		const { dns } = this.props;
+		if ( ! dns || ! dns.records ) {
+			return false;
+		}
+
+		return !! dns.records.filter( ( record ) => {
```

Comment:
Optional, this can be shortened to `dns.records.some( ( { data } ) => data === 'autodiscover.outlook.com.' );`